dom/events/scrolling/wheel-event-transactions-multiple-action-chains.tentative.html

```diff
@@ -0,0 +1,100 @@
+<!DOCTYPE HTML>
+<html>
+<head>
+<meta name="viewport" content="width=device-width,initial-scale=1,minimum-scale=1">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/resources/testdriver.js"></script>
+<script src="/resources/testdriver-actions.js"></script>
+<script src="/resources/testdriver-vendor.js"></script>
+<script src="scroll_support.js"></script>
+<style>
+body {
+  margin: 0;
+  padding: 0;
+  height: 200vh;
+}
+
+#initial {
+  background: red;
+  width: 100%;
+  height: 25px;
+  padding: 0;
+  margin: 0;
+}
+
+#second {
+  background: green;
+  width: 100%;
+  height: 200vh;
+  padding: 0;
+  margin: 0;
+}
+
+</style>
+<head>
+<body>
+<div id="initial"></div>
+<div id="second"></div>
+</body>
+
+<script>
+
+promise_test(async (t) => {
+  await new Promise(resolve => addEventListener("load", resolve, {once: true}));
+
+  await waitForCompositorReady();
+
+  await waitForCompositorCommit();
+
+  let firstEventTargets = [];
+  let secondEventTargets = [];
+  let isFirstGroup = true;
+  window.addEventListener("wheel", (e) => {
+    if (isFirstGroup) {
+      firstEventTargets.push(e.target);
+    } else {
+      secondEventTargets.push(e.target);
+    }
+  }, {passive: true});
+
+  await waitForCompositorCommit();
+
+  // The first action chain should target the initial element
+  await new test_driver.Actions()
+    .addWheel("wheel1")
+    .scroll(40, 2, 0, 30, {origin: "viewport"})
+    .send();
+
+  // Start logging event targets in the second transaction
+  isFirstGroup = false;
+
+  // The second chain should target the second element and the prior wheel event
+  // group should have no impact on this action chain.
+  await new test_driver.Actions()
+    .addWheel("wheel1")
```

Comment:
I'm not sure whether this causes another transaction in any conditions. How about to add a long `pause()`? Up to you.

Reply:
This is sort of a webdriver specific issue. A webdriver user would not expect two separate wheel action chains to have the same target, if the coordinates were different. A `pause()` test is valid, but the purpose of this test is to ensure that two separate action chains (no matter how much time lapses between them) always result in separate wheel transactions. @whimboo Do you have any other details to add here?

Reply:
Hm, I'm not familiar with wheel transactions as of now so not sure what to say here. At least you should not reuse the same input source id as long as the former action chain hasn't been released.

Beside that note that with the patch on [bug 1852243](https://bugzilla.mozilla.org/show_bug.cgi?id=1852243) we potentially will always have a new transaction whether how long the pause will be, right?

Reply:
This test is essentially written to test [bug 1821733](https://bugzilla.mozilla.org/show_bug.cgi?id=1821733).

> Beside that note that with the patch on [bug 1852243](https://bugzilla.mozilla.org/show_bug.cgi?id=1852243) we potentially will always have a new transaction whether how long the pause will be, right?

Assuming a new action chain, then yes we should always have a new wheel transaction for the new action chain.